### PR TITLE
feat: refactor the agency selection

### PR DESCRIPTION
**Pull request by CarlosSoares**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: src/components/radioButton/RadioButton.tsx [hunk 1]
```diff
@@ -3,7 +3,8 @@ import classNames from 'classnames';
 import './radioButton.styles';
 
 export interface RadioButtonItem {
-	checked: boolean;
+	checked?: boolean;
+	checkedValue?: boolean;
```
Comment (SheepFromHeaven) [inline src/components/radioButton/RadioButton.tsx]: shouldn't one of these rather be a state?

**Review by SheepFromHeaven — DISMISSED** [review]

File: src/components/radioButton/RadioButton.tsx (hunk 1 above)
Comment (CarlosSoares) [reply src/components/radioButton/RadioButton.tsx]: Fixed

File: src/components/mainTopicSelection/MainTopicSelection.tsx [hunk 2]
```diff
@@ -65,7 +65,7 @@ export const MainTopicSelection = ({
 								onChangeInput(id.toString())
 							}
 							value={id.toString()}
-							checked={selectedTopic === id}
+							checked={selectedTopic === id.toString()}
```
Comment (SheepFromHeaven) [inline src/components/mainTopicSelection/MainTopicSelection.tsx]: I see this "toString" everywhere. Can't we move this to the "apiGetTopicsData" function?

File: src/containers/registration/hooks/useConsultingTypeAgencySelection.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+			setAgencies(consultant?.agencies);
+			setConsultingTypes([consultingType]);
+			return;
+		}
+
+		const consultingTypes = consultant.agencies
+			// Remove consultingType duplicates
+			.reduce((acc: ConsultingTypeInterface[], { consultingTypeRel }) => {
+				if (
+					!acc.find(
```
Comment (SheepFromHeaven) [inline src/containers/registration/hooks/useConsultingTypeAgencySelection.ts]:
```suggestion
					!acc.some(
```

File: src/containers/registration/hooks/useConsultingTypeAgencySelection.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+						(consultingType) =>
+							consultingType.id === consultingTypeRel.id
+					)
+				) {
+					acc.push(consultingTypeRel);
```
Comment (SheepFromHeaven) [inline src/containers/registration/hooks/useConsultingTypeAgencySelection.ts:45]:
```suggestion
					return [...acc, consultingTypeRel];
```
I would reduce array manipulation to a minimum

File: src/containers/registration/hooks/useAgenciesForRegistration.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+		useConsultingTypeAgencySelection(
+			consultant,
+			consultingType,
+			preSelectedAgency
+		);
+	const [agencies, setAgencies] = useState<AgencyDataInterface[]>([]);
+	const { autoSelectPostcode, autoSelectAgency } =
+		consultingType?.registration || {};
+	const allAgencies = useMemo(() => {
+		let uniqueAgencies = uniqueBy(
```
Comment (SheepFromHeaven) [inline src/containers/registration/hooks/useAgenciesForRegistration.ts]:
```suggestion
		const uniqueAgencies = uniqueBy(
```

File: src/containers/registration/hooks/useAgenciesForRegistration.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+				propPreSelectedAgency,
+				...agencies,
+				...consultingTypeAgencies
+			].filter(Boolean),
+			'id'
+		);
+
+		if (consultingType && !autoSelectPostcode) {
+			// Hide external agencies in this case
+			uniqueAgencies = uniqueAgencies.filter(
```
Comment (SheepFromHeaven) [inline src/containers/registration/hooks/useAgenciesForRegistration.ts]:
```suggestion
			return uniqueAgencies.filter(
```

File: src/components/registration/RegistrationForm.tsx [hunk 7]
```diff
@@ -264,14 +188,13 @@ export const RegistrationForm = ({
...
-						preselectedAgencyData={preselectedAgencyData}
-						initialPostcode={initialPostcode}
+						preselectedAgencyData={agency}
 						onChange={handleChange}
 						additionalStepsData={consultingType?.requiredComponents}
 						registrationNotes={consultingType?.registration.notes}
 						consultant={consultant}
 						onValidation={setFormAccordionValid}
-						mainTopicId={formAccordionData.mainTopicId}
+						mainTopicId={topic?.id || formAccordionData.mainTopicId}
```
Comment (SheepFromHeaven) [inline src/components/registration/RegistrationForm.tsx:197]: shouldn't this be .toString to?

**Review by SheepFromHeaven — CHANGES_REQUESTED** [review]

**Review by tkuzynow — DISMISSED** [review]

**Review by RunzelRosinchen — DISMISSED** [review]

**Review by tkuzynow — APPROVED** [review]

**Review by SheepFromHeaven — APPROVED** [review]

**Pull request merged**